src/actions/ast.js

```diff
@@ -17,3 +17,18 @@ export function setSymbols(source) {
     });
   };
 }
+
+export function outOfScopeLocations() {
+  return async ({ dispatch, getState }: ThunkArgs) => {
+    const location = getSelectedLocation(getState());
+    const sourceText = getSourceText(getState(), location.sourceId);
```

Comment:
since we are using the same state object, maybe we can save it into a variable rather than calling it multiple times